src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py

```diff
@@ -7503,6 +7503,140 @@ def test_network_virtual_appliance_reimage(self, resource_group):
+            'name': 'cli-virtual-appliance-vnet',
+            'vendor': 'barracudasdwanrelease',
+            'version': 'latest',
+            'asn': '10000',
+            'scale': '2',
+        })
+
+        # -------- (1) vnet/subnet --------
+        self.cmd(
+            'network vnet create '
+            '-g {rg} -n {vnet} -l eastus2euap '
+            '--address-prefixes {addrSpace} '
+            '--subnet-name {subnetPriv} --subnet-prefix {privPrefix}'
+        )
+        self.cmd(
+            'network vnet subnet create '
+            '-g {rg} --vnet-name {vnet} -n {subnetPub} '
+            '--address-prefixes {pubPrefix}'
+        )
+        vnet = self.cmd('network vnet show -g {rg} -n {vnet}').get_output_in_json()
+        sub_priv_id = [s['id'] for s in vnet['subnets'] if s['name'] == self.kwargs['subnetPriv']][0]
+        sub_pub_id = [s['id'] for s in vnet['subnets'] if s['name'] == self.kwargs['subnetPub']][0]
+
+        # -------- (2) create interface JSON --------
+        iface_cfg_path = os.path.join(self.create_temp_dir(), 'nva-interface-config-example.json')
+        iface_cfg = [
+            {"name": "nic0", "subnet": {"id": sub_priv_id}, "type": ["PrivateNic"]},
+            {"name": "nic1", "subnet": {"id": sub_pub_id}, "type": ["PublicNic"]},
+        ]
+
+        with open(iface_cfg_path, 'w', encoding='utf-8') as f:
+            json.dump(iface_cfg, f, indent=2)
+        self.kwargs.update({'iface': iface_cfg_path})
+
+        # -------- (3) create nva --------
+        self.cmd(
+            'network virtual-appliance create '
+            '-n {name} -g {rg} '
+            '--vendor {vendor} --scale-unit {scale} -v {version} --asn {asn} '
+            '--interface-config "@{iface}"',
+            checks=[
+                self.check('name', '{name}'),
+                self.check('length(nvaInterfaceConfigurations)', 2),
+                self.check('virtualApplianceAsn', '{asn}')
+            ]
+        )
+
+        # -------- (4) loop until Succeeded --------
+        retry_count = 0
+        while True:
+            state = self.cmd('network virtual-appliance show -g {rg} -n {name}').get_output_in_json().get('provisioningState')
+            if state == 'Succeeded':
+                break
+            retry_count += 1
+            if retry_count >= 20:  # 20 * 60s ≈ 20 min
+                raise Exception(f'Operation not complete: provisioningState={state}, retry_count={retry_count}')
+            sleep(60)
+
+        # -------- (5) wait 15 min --------
+        print('Provisioning Succeeded → sleep 5 minutes before update')
+        sleep(900)
+
+        # -------- (6) attempt update; if it fails, retry 5 times, sleeping for 10 minutes before each attempt --------
+        max_retries = 5
+        attempt = 0
+        last_err = None
+        while True:
+            try:
+                self.cmd(
+                    'network virtual-appliance update -n {name} -g {rg} --asn 20000',
```

Comment:
our targer parameter is `--interface-config` (not `asn`), seems we didn't cover it in the update testing cmd.